### PR TITLE
Feature/admin list outstanding balances

### PR DESCRIPTION
**Pull request by gearsix**
- Added a Member class to .models
- Moved MembersService functionality to MembersRepo
- Added list outstanding balances
- Updated adminController & admin.jsp to reflect membersRepo functions
- Updated SQLService to make it more useable (made sure I didn't break anything using it).

I had to add a function overload to getWhere() in the abstract Repo class for listing outstanding members.

### CODE REVIEW

File: XYZDrivers/src/main/java/com/xyzdrivers/controllers/AdminController.java [hunk 1]
```diff
@@ -32,15 +32,21 @@
...
     {
-        /* create Connection to DB */
+        //create Connection to DB
         Class.forName("com.mysql.jdbc.Driver");
-        SQLService sql = new SQLService(DriverManager.getConnection("jdbc:derby://localhost:1527/xyzdrivers", "root", "root"));
-        
-        List<Object[]> members = MembersService.getMembers(sql);
+        Connection conn = DriverManager.getConnection("jdbc:derby://localhost:1527/xyzdrivers", "root", "root");
+        //create service & repo instances
+        MembersRepo membersRepo = new MembersRepo();
```
Comment (bkneis) [inline XYZDrivers/src/main/java/com/xyzdrivers/controllers/AdminController.java]: This should be injected

File: XYZDrivers/src/main/java/com/xyzdrivers/controllers/AdminController.java [hunk 2]
```diff
@@ -32,15 +32,21 @@
     protected void processRequest(HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException, SQLException, ClassNotFoundException
     {
-        /* create Connection to DB */
+        //create Connection to DB
         Class.forName("com.mysql.jdbc.Driver");
-        SQLService sql = new SQLService(DriverManager.getConnection("jdbc:derby://localhost:1527/xyzdrivers", "root", "root"));
-        
-        List<Object[]> members = MembersService.getMembers(sql);
+        Connection conn = DriverManager.getConnection("jdbc:derby://localhost:1527/xyzdrivers", "root", "root");
```
Comment (bkneis) [inline XYZDrivers/src/main/java/com/xyzdrivers/controllers/AdminController.java]: No need for this variable with the repo

File: XYZDrivers/src/main/java/com/xyzdrivers/services/SQLService.java [hunk 3]
```diff
@@ -235,94 +273,60 @@ public boolean exists(String table, String column, Object item)
...
-     * @throws IllegalArgumentException if <code>(query results).next()</code> returns false.
-     * @throws SQLException 
-     */
-    public Object retrieve(String table, String column, String primaryKeyColumn, Object primaryKey)
-            throws SQLException, IllegalArgumentException
-    {
         //execute statement
-        results = executeQueryStatement("SELECT "+column+" FROM "+table+" WHERE "+primaryKeyColumn+" = ?", primaryKey);
-        
+        results = executeQueryStatement("SELECT "+column+" FROM "+table+" WHERE \""+keyColumn+"\" = ?", keyValue);
```
Comment (bkneis) [inline XYZDrivers/src/main/java/com/xyzdrivers/services/SQLService.java]: please add spaces after +

File: XYZDrivers/src/main/java/com/xyzdrivers/services/SQLService.java [hunk 4]
```diff
@@ -235,94 +273,60 @@ public boolean exists(String table, String column, Object item)
...
      * 
      * @throws SQLException 
      */
-    public void update(String table, String column, String primaryKeyColumn, Object primaryKey, Object value)
+    public void update(String table, String column, Object value, String keyColumn, Object keyValue)
             throws SQLException
     {
         //execute statement
-        executeUpdateStatement("UPDATE "+table+" SET "+column+" = ? WHERE "+primaryKeyColumn+" = ?", value, primaryKey);
+        executeUpdateStatement("UPDATE "+table+" SET "+column+" = ? WHERE "+keyColumn+" = ?", value, keyValue);
```
Comment (bkneis) [inline XYZDrivers/src/main/java/com/xyzdrivers/services/SQLService.java]: spaces

**Review by bkneis — COMMENTED** [review]
Nice, well documented. Just a couple of minor things

**Comment by bkneis** [comment]
Thank you, but you left out removing the redundant connection variable and also injecting the repo

**Comment by gearsix** [comment]
Hey, changes in the review should be good now. Couldn't find the redundant connection in AdminController though, thought I'd removed it anyway.

**Review by bkneis — APPROVED** [review]
LGTM. Cheers for the changes

**Pull request merged**